pennylane/interfaces/execution.py

```diff
@@ -758,13 +761,6 @@ def device_gradient_fn(inner_tapes, **gradient_kwargs):
 
             gradient_fn = device_gradient_fn
 
-            # Adjoint Jacobian with backward pass and jitting needs the original circuit output state which
-            # can not be reused from the device if `grad_on_execution is False`.
-            if interface == "jax-jit":
-                use_device_state = gradient_kwargs.get("use_device_state", None)
-                if use_device_state:
-                    gradient_kwargs["use_device_state"] = False

```

Comment:
is this logic no longer used? seems like it's relevant for `QubitDevice.adjoint_jacobian`

Reply:
We are now in a situation where we either just calculate the results `cost(x)` or we always calculate the results and gradients at the same time `jax.jacobian(cost)(x)`. We no longer have the case where we do the forward pass, leave the pure callback, and then come back and calculate the jacobian later. So we could technically just set `use_device_state=True` for all cases.

Reply:
@albi3ro since this this looks like interface-agnostic code, what about other interfaces, like PyTorch? E.g.,

```python
res = cost(x)
... # more computation
# later
res.backward()
```

Here, unless something else has changed, we have a case where the forward pass occurs, and then later the gradients are requested.

Reply:
This was a fix specific to JAX (see removed line 763).